### PR TITLE
#155: Optimize atomium_theme_registry_alter()

### PR DESCRIPTION
**Pull request by donquixote**
@drupol Do not merge yet! This branch contains some "preview" junk.

I measured this and got measurable improvement for almost every optimization step.

Overall:
before: 1.9140625
after: 1.1630859375

for this specific function.

Measured via a hack in drupal_alter():

```diff
diff --git a/includes/module.inc b/includes/module.inc
index 4c2b3fb..f1d6830 100644
--- a/includes/module.inc
+++ b/includes/module.inc
@@ -1168,6 +1168,21 @@ function drupal_alter($type, &$data, &$context1 = NULL, &$context2 = NULL, &$con
   }
   foreach ($functions[$cid] as $function) {
-    $function($data, $context1, $context2, $context3);
+    if ('atomium_theme_registry_alter' === $function) {
+      $data_before = $data;
+      $dtsMs = [];
+      for ($i = 0; $i < 10; ++$i) {
+        $data = $data_before;
+        $t0Ms = microtime(TRUE) * 1000;
+        $function($data, $context1, $context2, $context3);
+        $t1Ms = microtime(TRUE) * 1000;
+        $dtsMs[] = $t1Ms - $t0Ms;
+      }
+      sort($dtsMs);
+      dpm($dtsMs[0], __FUNCTION__ . '() calling ' . $function . '()');
+    }
+    else {
+      $function($data, $context1, $context2, $context3);
+    }
   }
 }
```

### CODE REVIEW

**Comment by donquixote** [comment]
Having the separate commits allows to measure every version, like this:

git checkout 7.x-3.x-local
git checkout 7.x-3.x-local^
git checkout 7.x-3.x-local^^
git checkout 7.x-3.x-local^^^
[..]
git checkout 7.x-3.x-local^^^^^^^^^^
etc
(here using my local branch)

File: atomium/includes/registry.inc [hunk 1]
```diff
@@ -70,35 +91,72 @@ function atomium_theme_registry_alter(array &$registry) {
...
-        $components[$base_hook]
-      )
-    );
-
-    // Remove 'atomium preprocess' callback.
-    $info['preprocess functions'] = array_filter(
-      $info['preprocess functions'],
-      function ($preprocess) {
-        return 'atomium_preprocess' !== $preprocess;
+  foreach ($registry as $hook => &$info) {
```
Comment (drupol) [inline atomium/includes/registry.inc:94]: :disappointed:

**Comment by donquixote** [comment]
I did this in a dedicated commit and measured the difference. Small but
measurable.

On Sep 14, 2018 20:34, "Pol Dellaiera" <notifications@github.com> wrote:

*@drupol* commented on this pull request.
------------------------------

In atomium/includes/registry.inc
<https://github.com/ec-europa/atomium/pull/156#discussion_r217804930>:

> -    $base_hook = atomium_get_base_hook($hook);
-    $components += array($base_hook => array());
-
-    $info['includes'] = array_unique(
-      array_merge(
-        $info['includes'],
-        $components[$base_hook]
-      )
-    );
-
-    // Remove 'atomium preprocess' callback.
-    $info['preprocess functions'] = array_filter(
-      $info['preprocess functions'],
-      function ($preprocess) {
-        return 'atomium_preprocess' !== $preprocess;
+  foreach ($registry as $hook => &$info) {

😞

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ec-europa/atomium/pull/156#pullrequestreview-155613525>,
or mute the thread

**Comment by drupol** [comment]
What's the status of this ?

**Comment by drupol** [comment]
Close in favor of #204

**Pull request closed without merging**